qt/python/mantidqt/widgets/common/observing_view.py

```diff
@@ -0,0 +1,55 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2019 ISIS Rutherford Appleton Laboratory UKRI,
+#     NScD Oak Ridge National Laboratory, European Spallation Source
+#     & Institut Laue - Langevin
+# SPDX - License - Identifier: GPL - 3.0 +
+#  This file is part of the mantid workbench.
+
+
+class ObservingView:
+    """
+    This class provides some common functions needed across views observing the ADS.
+    It runs the close_signal so that the view is closed from the GUI thread,
+    and ensures that the closeEvent will clear the observer to prevent a memory leak.
+
+    It is designed to be used with a presenter that inherits `ObservingPresenter`.
+
+    If this class is inherited a `close_signal` and a `rename_signal` must be declared.
+
+    It is not possible to do that here, as this is not a QObject, however it was
+    attempted to do the declaration here, but the signals don't seem to work
+    through inheritance.
+
+    This class shouldn't inherit from a QObject/QWidget, otherwise the closeEvent
+    doesn't replace the closeEvent of the view that is inheriting this. This also
+    makes it easily testable, as the GUI library isn't pulled in.
+    """
+
+    TITLE_STRING = "{} - Mantid"
+
+    def __init__(self, _):
```

Comment:
It's not immediately obvious to why we need the `__init__`? I assume I've missed something so can we add a comment?

Reply:
I thought I needed it because of the view's `super(..., self).__init__(parent)` call, which calls both the Qt constructor, and then this one, but it turns out there's no RuntimeError if I remove it. 

I would prefer to leave it in though, as it removes a `Unresolved attribute reference '__init__' for class 'ObservingView'` warning in PyCharm, which might be confusing in its own right.

Reply:
That's a fair reason. :+1: for the comment